PatternPal/PatternPal.Core/Recognizers/Helper_Classes/AdapterRecognizerParent.cs

```diff
@@ -0,0 +1,207 @@
+﻿#region
+using PatternPal.SyntaxTree.Models;
+using static PatternPal.Core.Checks.CheckBuilder;
+#endregion
+
+namespace PatternPal.Core.Recognizers.Helper_Classes;
+
+/// <summary>
+/// An abstract class which contains all checks for the adapter<see cref="IRecognizer"/>.
+/// Because the adapter-pattern can be applied using an interface or an abstract class, some methods are abstract and specified in the subclass.
```

Comment:
I would make this more general and instead of "some methods are abstract and specified in the subclass" change it to: "some requirements are interface or abstract class specific and defined in the subclasses"